core/graphelier-service/db/mongo.go

```diff
@@ -29,20 +28,64 @@ func NewConnection() (*Connector, error) {
 	if err = client.Ping(context.TODO(), nil); err != nil {
 		log.Fatal(err)
 	}
-	fmt.Println("Connected to MongoDB :)")
+	log.Println("Connected to MongoDB :)")
 
 	return &Connector{client}, nil
 }
 
-// FindOrderbook : Finds the Orderbook of an instrument based on the timestamp requested in the db
-func (c *Connector) FindOrderbook(instrument string, timestamp float64) (result *models.Orderbook) {
+// GetOrderbook : Finds the Orderbook of an instrument based on the timestamp requested in the db
+func (c *Connector) GetOrderbook(instrument string, timestamp uint64) (result *models.Orderbook) {
 	collection := c.Database("graphelier-db").Collection("orderbooks")
-	filter := bson.D{{Key: "instrument", Value: instrument}, {Key: "timestamp", Value: timestamp}}
+	filter := bson.D{
+		{Key: "instrument", Value: instrument},
+		{Key: "timestamp", Value: bson.D{
+			{Key: "$lte", Value: timestamp},
+		}},
+	}
+	option := options.FindOne()
+	option.SetSort(bson.D{{Key: "timestamp", Value: -1}})
 
-	err := collection.FindOne(context.TODO(), filter).Decode(&result)
+	err := collection.FindOne(context.TODO(), filter, option).Decode(&result)
 	if err != nil {
 		log.Fatal(err)
 	}
```

Comment:
seems repetitive in code, could be a function call ?

Reply:
This is how exception handling is done in go. Can't really be refactored